pkg/runtime/reconciler.go

```diff
@@ -152,6 +153,43 @@ func (r *reconciler) SecretValueFromReference(
 	return "", ackerr.SecretNotFound
 }
 
+// WriteToSecret writes a value to a Secret given the namespace, name,
+// and key of the Secret
+func (r *reconciler) WriteToSecret(
+	ctx context.Context,
+	sourceValue string,
+	namespace string,
+	name string,
+	key string,
+) error {
+
+	// Get the initial secret
+	nsn := types.NamespacedName{
+		Name: name,
+	}
+	nsn.Namespace = namespace
+
+	secret := &corev1.Secret{}
+	err := r.apiReader.Get(ctx, nsn, secret)
+	if err != nil {
+		return ackerr.SecretNotFound
+	}
```

Comment:
I think we don't need to do the `Get` here. Patch will fail anyways if the secret doesn't exist. Can you test this function with no `Get` calls?

Reply:
`Get` retrieves the secret from the given namespace and name so I think we need to call it before Patch. I followed the field export WriteToSecret method [https://github.com/aws-controllers-k8s/runtime/blob/e5668082edc3bf40d7d2d6bdcf43e0189426d528/pkg/runtime/field_export_reconciler.go#L336C1-L382C1](https://github.com/aws-controllers-k8s/runtime/blob/e5668082edc3bf40d7d2d6bdcf43e0189426d528/pkg/runtime/field_export_reconciler.go#L336C1-L382C1)

Reply:
Makes sense, thank you!

Reply:
One last ask, can we reuse this function in fieldExportReconciler?

Reply:
Discussed with Neha off-line we will address this in a seperate PR before releasing runtime 0.34.0